pkg/scheduler/core/generic_scheduler.go

```diff
@@ -488,8 +491,15 @@ func podFitsOnNode(
 						} else {
 							predicateResults[predicateKey] = HostPredicate{Fit: fit, FailReasons: reasons}
 						}
+						result := predicateResults[predicateKey]
+						ecache.UpdateCachedPredicateItem(pod.GetName(), info.Node().GetName(), predicateKey, result.Fit, result.FailReasons, equivCacheInfo.hash, false)
 					}
 				}
+
+				if eCacheAvailable {
+					ecache.Unlock()
```

Comment:
Why not release lock just following `UpdateCachedPredicateItem()`? 

Also, I am not sure about purpose of `needLock bool`, seems like you can just move lock control out from ecache.

You may want to still use defer to ensure  `ecache.Unlock()`.

Reply:
>Emm ... why not release lock just following UpdateCachedPredicateItem() above?

Lock from `ecache.PredicateWithECache` to `UpdateCachedPredicateItem` to ensure that the result of `PredicateWithECache` is not changed when update the ecache.

>Also, I am not sure about purpose of needLock bool, seems like you can just move lock control out from ecache.

I just suppose there may be requirements in future for using ` UpdateCachedPredicateItem()` with lock inside the method.

@resouer